File: public/cgi-bin/test_cgi.py
#!/usr/bin/env python3
import os
import sys

print("Content-Type: text/plain\n")

method = os.environ.get("REQUEST_METHOD", "")
if method == "GET":
    print("Received GET")
    print("Query:", os.environ.get("QUERY_STRING", ""))
elif method == "POST":
    length = int(os.environ.get("CONTENT_LENGTH", 0))
    post_data = sys.stdin.read(length)
    print("Received POST")
    print("Data:", post_data)
sys.stdout.flush()

File: public/cgi-bin/test_cgi.php
#!/usr/bin/env php
<?php
// test_cgi.php: Handles GET and POST requests via CGI.
// - For GET: prints method and query string.
// - For POST: reads and prints the raw POST body.

// Print the CGI header manually
echo "Content-Type: text/plain\r\n\r\n";

$method = $_SERVER["REQUEST_METHOD"];

if ($method === "GET") 
{
    echo "Received GET\n";
    echo "Query: " . ($_SERVER["QUERY_STRING"] ?? "") . "\n";
} 
elseif ($method === "POST") 
{
	$stdin = fopen('php://stdin', 'r');
    $body = stream_get_contents($stdin);
    echo "Received POST\n";
    echo "Data: $body\n";
}
?>
